example-tests/src/test/java/com/epam/restassured/csvreader/CSVReaderUtilitySingleton.java

```diff
@@ -0,0 +1,105 @@
+package com.epam.restassured.csvreader;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.epam.restassured.pojo.csv.CSVTestInput;
+import com.opencsv.CSVReader;
+
+/**
+ * Singleton representation of a CSV Reader utility class.
+ * 
+ * Please note, CSV input files should follow the following format:
+ * No comment line, data should start in the first line in the following oder
+ * FirstName,LastName,EmailAddress,EmailAddressConfirmation,NewsletterOptIn
+ * 
+ * Open issue for OpenCSV to add comment line in the first line of the input CSV file
+ * 
+ * @author Tamas_Csako
+ *
+ */
+public class CSVReaderUtilitySingleton {
```

Comment:
It's unnecessary to be singleton. Its instantiation cost is nearly nothing, the costly file reading is defined in an instance method.